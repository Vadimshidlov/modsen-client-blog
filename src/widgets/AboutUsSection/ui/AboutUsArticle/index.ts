export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
