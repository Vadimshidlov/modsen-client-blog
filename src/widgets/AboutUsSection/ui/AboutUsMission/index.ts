export { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission/AboutUsMission';
