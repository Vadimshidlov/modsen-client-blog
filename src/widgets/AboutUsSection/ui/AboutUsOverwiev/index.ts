export { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev/AboutUsOverwiev';
