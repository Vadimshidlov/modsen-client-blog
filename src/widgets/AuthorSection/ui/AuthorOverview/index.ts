export { AuthorOverview } from '@/widgets/AuthorSection/ui/AuthorOverview/AuthorOverview';
