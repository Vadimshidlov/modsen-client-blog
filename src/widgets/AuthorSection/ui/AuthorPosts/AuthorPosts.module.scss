.authorPostsSection {
  display: flex;
  flex-direction: column;
  margin: 0px auto 128px;
  padding-top: 128px;
  max-width: 1024px;
  padding-right: 20px;
  padding-left: 20px;

  @media (max-width: $tablet) {
    padding-top: 32px;
    margin-bottom: 64px;
  }

  @media (max-width: $mobile) {
    padding-top: 16px;
    margin-bottom: 32px;
  }

  .sectionTitle {
    margin-bottom: 64px;
    @include headingOne;

    @media (max-width: $tablet) {
      margin-bottom: 32px;
      @include headingThree;
    }

    @media (max-width: $mobile) {
      margin-bottom: 16px;
      @include headingFour;
    }
  }

  .authorPostsContainer {
    display: flex;
    flex-direction: column;
    row-gap: 64px;

    @media (max-width: $tablet) {
      row-gap: 32px;
    }

    @media (max-width: $mobile) {
      row-gap: 16px;
    }
  }
}
