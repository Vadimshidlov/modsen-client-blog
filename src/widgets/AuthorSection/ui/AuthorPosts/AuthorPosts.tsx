import React from 'react';
import { POSTS } from '@/shared/constants/posts';
import { PostItem } from '@/widgets/PostItem';
import { useTranslations } from 'next-intl';
import { AuthorPostsPropsType } from '@/shared/types';
import styles from '@/widgets/AuthorSection/ui/AuthorPosts/AuthorPosts.module.scss';

export function AuthorPosts({ authorName }: AuthorPostsPropsType) {
  const t = useTranslations('AuthorPosts');
  const authorPosts = POSTS.filter((postData) => postData.author === authorName);

  return (
    <section className={styles.authorPostsSection}>
      <h1 className={styles.sectionTitle}>{t('title')}</h1>
      <article className={styles.authorPostsContainer}>
        {authorPosts.map(({ id, title, img, category, text }) => (
          <PostItem id={id} text={text} title={title} category={category} img={img} />
        ))}
      </article>
    </section>
  );
}
