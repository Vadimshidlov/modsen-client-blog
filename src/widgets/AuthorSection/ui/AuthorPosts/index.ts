export { AuthorPosts } from '@/widgets/AuthorSection/ui/AuthorPosts/AuthorPosts';
