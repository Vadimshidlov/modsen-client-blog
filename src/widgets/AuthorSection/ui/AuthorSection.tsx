import React from 'react';
import { AuthorOverview } from '@/widgets/AuthorSection/ui/AuthorOverview';
import { AuthorPosts } from '@/widgets/AuthorSection/ui/AuthorPosts';
import styles from '@/widgets/AuthorSection/ui/AuthorSection.module.scss';
import { AuthorSectionPropsType } from '@/shared/types';

export function AuthorSection({ name, img }: AuthorSectionPropsType) {
  return (
    <div className={styles.authorSection}>
      <AuthorOverview name={name} img={img} />
      <AuthorPosts authorName={name} />
    </div>
  );
}
