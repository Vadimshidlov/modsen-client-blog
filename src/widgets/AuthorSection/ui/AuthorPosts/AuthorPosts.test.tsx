import { render } from '@testing-library/react';
import { AuthorPosts } from '@/widgets/AuthorSection/ui/AuthorPosts';

jest.mock('next/font/google', () => ({
  Inter: () => ({ variable: 'fake-variable' }),
}));

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('AboutUsTitle tests', () => {
  const name = 'John Doe';

  test('should matches snapshot', () => {
    const { container } = render(<AuthorPosts authorName={name} />);
    expect(container).toMatchSnapshot();
  });
});
