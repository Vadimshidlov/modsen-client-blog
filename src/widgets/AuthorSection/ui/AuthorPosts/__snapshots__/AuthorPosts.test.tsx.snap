// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUsTitle tests should matches snapshot 1`] = `
<div>
  <section
    class="authorPostsSection"
  >
    <h1
      class="sectionTitle"
    >
      Mocked title
    </h1>
    <article
      class="authorPostsContainer"
    />
  </section>
</div>
`;
